torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -147,23 +145,41 @@ def __init__(
         parameters_as_bucket_view: bool = False,
         **default: Any,
     ):
+        # Validate the type and length of `params`
+        # NOTE: While `Optimizer` allows `params` to be an iterable of `dict`s,
+        # ZeRO strictly requires `params` to be an iterable of `torch.Tensor`s
+        if (isinstance(params, torch.Tensor)):
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
+        try:
+            self._all_params = list(params)
+        except TypeError:
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
+        if (len(self._all_params) == 0):
+            raise ValueError("ZeroRedundancyOptimizer got an empty parameter "
+                             "list")
+        if (not isinstance(self._all_params[0], torch.Tensor)):
```

Comment:
do we need to check all params instead of the first one?

Reply:
Fixed.